dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMain.java

```diff
@@ -60,25 +60,73 @@ public static void main(String[] arguments)
         // If JAR files are specified, it's empty.
         ImmutableList<Artifact> artifacts = linkageCheckerArguments.getArtifacts();
 
-        ImmutableSet<LinkageProblem> linkageProblems =
+        Problems problems =
             artifacts.isEmpty()
                 ? checkJarFiles(linkageCheckerArguments)
                 : checkArtifacts(linkageCheckerArguments);
-        if (!linkageProblems.isEmpty()) {
-          System.out.println(
-              "For the details of the linkage errors, see "
-                  + "https://github.com/GoogleCloudPlatform/cloud-opensource-java/wiki/Linkage-Checker-Messages");
-          // Throwing an exception is more test-friendly compared with System.exit(1). The latter
-          // abruptly stops test execution.
-          throw new LinkageCheckResultException(linkageProblems.size());
+
+        Path outputExclusionFile = linkageCheckerArguments.getOutputExclusionFile();
+        // All I/O happens here
+        if (!problems.linkageProblems.isEmpty()) {
+          // TODO really uncertain about this check. Whether to write an exclusion file is
+          // a separate issue from whether to print the linkage problems.
+          if (outputExclusionFile == null) {
```

Comment:
Current check looks good. I feel the similarity with `curl -o <filename>`. If we chose to write the output to the file, it does not print the information on console.